perl.c

```diff
@@ -2548,6 +2550,8 @@ S_parse_body(pTHX_ char **env, XSINIT_t xsinit)
 	filter_add(read_e_script, NULL);
 
     /* now parse the script */
+    if (minus_e == FALSE)
+        PL_hints |= HINT_DEFAULT; /* after init_main_stash ; need to be after init_predump_symbols */
```

Comment:
Currently perl programs rely on PL_hints being set to 0 by default. As a result there is no current code that initializes it.